File: shop-app-server/src/main/java/com/medochek/shopapp/domain/Product.java
package com.medochek.shopapp.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Table
@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
    private String description;
    private String image;
    private Double price;
}

File: shop-app-web-user/src/main/web/src/app/models/product.ts
export class Product {
    private _id: number;
    private _name: string;
    private _price: number;
    private _description: string;
    private _photo: string;


    constructor(data: any) {
        this._id = data.id;
        this._name = data.name;
        this._price = data.price;
        this._description = data.description;
        this._photo = data.image;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get price(): number {
        return this._price;
    }

    set price(value: number) {
        this._price = value;
    }

    get description(): string {
        return this._description;
    }

    set description(value: string) {
        this._description = value;
    }

    get photo(): string {
        return this._photo;
    }

    set photo(value: string) {
        this._photo = value;
    }
}